### PR TITLE
Update APIServer to make room for audit profile

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

**Comment by tkashem** [comment]
work in progress

/hold

**Comment by tkashem** [comment]
ready for review 

/hold cancel

File: config/v1/types_apiserver.go [hunk 1]
```diff
@@ -51,6 +51,37 @@ type APIServerSpec struct {
 	// is VersionTLS12.
 	// +optional
 	TLSSecurityProfile *TLSSecurityProfile `json:"tlsSecurityProfile,omitempty"`
+	// audit specifies the settings for audit configuration to be applied to all OpenShift-provided
+	// API servers in the cluster.
+	Audit Audit `json:"audit,omitempty"`
```
Comment (sttts) [inline config/v1/types_apiserver.go:56]: remove omitempty. This has no effect.

File: config/v1/types_apiserver.go [hunk 2]
```diff
@@ -51,6 +51,37 @@ type APIServerSpec struct {
...
+type Audit struct {
+	// profile specifies the name of the desired audit policy configuration to be deployed to
+	// all OpenShift-provided API servers in the cluster
+	//
+	// We provide the following profiles
+	// - Default
+	// - WriteRequestBodies
+	// - AllRequestBodies
+	// If unset, the 'Default' profile is used as the default.
+	Profile AuditProfileType `json:"profile,omitempty"`
```
Comment (sttts) [inline config/v1/types_apiserver.go:84]: add kubebuilder marker for a default `"Default"`

Comment (sttts) [reply config/v1/types_apiserver.go:84]: plus an enum marker for the three values

Comment (sttts) [reply config/v1/types_apiserver.go:84]: https://book.kubebuilder.io/reference/markers/crd-validation.html

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (sttts) [reply config/v1/types_apiserver.go:56]: We also need a default marker here afaik, with `{"profile":"Default"}`

File: config/v1/types_apiserver.go (hunk 2 above)
Comment (sttts) [reply config/v1/types_apiserver.go:88]: > plus an enum marker for the three values

still missing

Comment (tkashem) [reply config/v1/types_apiserver.go:88]: @sttts I have put it where the enum is declared - https://github.com/openshift/api/pull/681/files#diff-2a98fd5f6a0f42f75cdc7d0bfbeb4333R62-R63


you want me to move it here?

Comment (sttts) [reply config/v1/types_apiserver.go:88]: Nope. You way is even better :+1:

File: config/v1/0000_10_config-operator_01_apiserver.crd.yaml [hunk 3]
```diff
@@ -53,6 +53,25 @@ spec:
...
+              type: object
+              default:
+                profile: Default
+              properties:
+                profile:
+                  description: "profile specifies the name of the desired audit policy
+                    configuration to be deployed to all OpenShift-provided API servers
+                    in the cluster \n We provide the following profiles - Default
+                    - WriteRequestBodies - AllRequestBodies If unset, the 'Default'
+                    profile is used as the default."
```
Comment (sttts) [inline config/v1/0000_10_config-operator_01_apiserver.crd.yaml:68]: how about specifying them a bit more?

**Pull request merged**